pkg/resource/processing_job/testdata/test_suite.yaml

```diff
@@ -1,17 +1,174 @@
+        description: "Testing readOne after processing job completes, the status should have ARN."
+        given:
+          desired_state: "v1alpha1/readone/desired/after_create_variation.yaml"
+          svc_api:
+            - operation: DescribeProcessingJobWithContext
+              output_fixture: "sdkapi/describe/success_describe_variation.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/completed_variation.yaml"
+      - name: "ReadOne=Stopping"
+        description: "Testing readOne right after delete, the status should be in Stopping"
+        given:
+          desired_state: "v1alpha1/readone/desired/stopping.yaml"
+          svc_api:
+            - operation: DescribeProcessingJobWithContext
+              output_fixture: "sdkapi/describe/stopping_describe.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/stopping.yaml"
+  - name: "Processing job update tests"
+    description: "Testing the Update operation"
+    scenarios:
+      - name: "Update=NotSupported"
+        description: "This test checks if the controller throws error for update"
+        given:
+          desired_state: "v1alpha1/update/desired/updated_base.yaml"
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/error_on_update.yaml"
+          error: "not implemented"
+  - name: "Processing job delete tests"
+    description: "Testing the delete operation"
+    scenarios:
+      - name: "Delete=Fail"
+        description: "This test checks if the condition is updated if delete fails and returns error"
+        given:
+          desired_state: "v1alpha1/readone/desired/after_create.yaml"
+          svc_api:
+            - operation: StopProcessingJobWithContext
+              error:
+                code: ServiceUnavailable
+                message: "Server is down"
+        invoke: Delete
+        expect:
+          latest_state: "v1alpha1/delete/observed/error_on_delete.yaml"
+          error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+      - name: "Delete=Inprogress"
+        description: "This test checks if the Processing job is deleted successfully"
+        given:
+          desired_state: "v1alpha1/readone/desired/after_create.yaml"
+          svc_api:
+            - operation: StopProcessingJobWithContext
+            - operation: DescribeProcessingJobWithContext
+              error:
+                code: ValidationException
+                message: "Could not find requested job"
+        invoke: Delete
+        expect:
+          error: nil      
+      - name: "Delete=Successful"
+        description: "This test checks if the Processing job returns nil to mark the resource Unmanaged if job not in InProgress state"
+        given:
+          desired_state: "v1alpha1/readone/observed/completed.yaml"
```

Comment:
Shouldn't this be tested on a job which is in-progress ?

Reply:
changed the test names

Reply:
thanks.